node/src/node.rs

```diff
@@ -66,14 +77,133 @@ where
+        Ok(self.p2p.listeners().await?)
+    }
+
+    /// Get all the peers that node is connected to.
+    pub async fn connected_peers(&self) -> Result<Vec<PeerId>> {
+        Ok(self.p2p.connected_peers().await?)
+    }
+
+    /// Trust or untrust the peer with a given ID.
+    pub async fn set_peer_trust(&self, peer_id: PeerId, is_trusted: bool) -> Result<()> {
+        Ok(self.p2p.set_peer_trust(peer_id, is_trusted).await?)
+    }
+
+    /// Request the head header from the network.
+    pub async fn request_head_header(&self) -> Result<ExtendedHeader> {
+        Ok(self.p2p.get_head_header().await?)
+    }
+
+    /// Request a header for the block with a given hash from the network.
+    pub async fn request_header_by_hash(&self, hash: &Hash) -> Result<ExtendedHeader> {
+        Ok(self.p2p.get_header(*hash).await?)
+    }
+
+    /// Request a header for the block with a given height from the network.
+    pub async fn request_header_by_height(&self, hash: u64) -> Result<ExtendedHeader> {
+        Ok(self.p2p.get_header_by_height(hash).await?)
+    }
+
+    /// Request headers in range (from, from + amount] from the network.
+    ///
+    /// The headers will be verified with the `from` header.
+    pub async fn request_verified_headers(
+        &self,
+        from: &ExtendedHeader,
+        amount: u64,
+    ) -> Result<Vec<ExtendedHeader>> {
+        Ok(self.p2p.get_verified_headers_range(from, amount).await?)
+    }
+
+    /// Get current header syncing info.
+    pub async fn syncer_info(&self) -> Result<SyncingInfo> {
+        Ok(self.syncer.info().await?)
+    }
+
+    /// Get the latest header announced in the network.
+    pub fn get_network_head_header(&self) -> Option<ExtendedHeader> {
+        self.p2p.header_sub_watcher().borrow().clone()
+    }
+
+    /// Get the latest locally synced header.
+    pub async fn get_local_head_header(&self) -> Result<ExtendedHeader> {
+        Ok(self.store.get_head().await?)
+    }
+
+    /// Get a synced header for the block with a given hash.
+    pub async fn get_header_by_hash(&self, hash: &Hash) -> Result<ExtendedHeader> {
+        Ok(self.store.get_by_hash(hash).await?)
+    }
+
+    /// Get a synced header for the block with a given height.
+    pub async fn get_header_by_height(&self, hash: u64) -> Result<ExtendedHeader> {
+        Ok(self.store.get_by_height(hash).await?)
+    }
+
+    /// Get synced headers from the given heights range.
+    ///
+    /// If start of the range is unbounded, the first returned header will be of height 1.
+    /// height.
+    /// If end of the range is unbounded, the last returned header will be the last header in the
+    /// store.
+    ///
+    /// # Errors
+    ///
+    /// If range contains a height of a header that is not found in the store or [`RangeBounds`]
+    /// cannot be converted to a valid range.
+    pub async fn get_verified_headers<R>(&self, range: R) -> Result<Vec<ExtendedHeader>>
```

Comment:
Why do we need `verified` in the name? Is it to avoid confusing the users?

Reply:
we don't need it, I have it kind of symmetrical to the `request_verified_headers` but now with the signature also different, I can rename it. up to you again

Reply:
removed _verified